src/typeInference/annotateSimpleMapExpr.ts

```diff
@@ -8,13 +13,24 @@ import { AnnotationContext } from './AnnotationContext';
  * @param ast the AST to be annotated.
  * @returns the inferred SequenceType
  */
-export function annotateSimpleMapExpr(ast: IAST, context: AnnotationContext): SequenceType {
-	// const seqType = {
-	// 	type: ValueType.MAP,
-	// 	mult: SequenceMultiplicity.EXACTLY_ONE,
-	// };
-
-	// astHelper.insertAttribute(ast, 'type', seqType);
-	// return seqType;
-	return undefined;
+export function annotateSimpleMapExpr(
+	ast: IAST,
+	context: AnnotationContext,
+	annotate: (ast: IAST, context: AnnotationContext) => SequenceType
```

Comment:
Since we're not passing annotate to any of the other annotation functions, wouldn't it be better to annotate all the pathExpr nodes before calling this function and then pass those as arguments. That way, its more consistent with the rest of the codebase.

Reply:
yup